vms/rpcchainvm/ghttp/http_test.go

```diff
@@ -49,3 +61,44 @@ func TestConvertWriteResponse(t *testing.T) {
 		})
 	}
 }
+
+func TestRequestClientArbitrarilyLongBody(t *testing.T) {
+	require := require.New(t)
+
+	listener := bufconn.Listen(0)
+	server := grpc.NewServer()
+	test.RegisterFooServer(server, fooServer{})
+
+	go func() {
+		require.NoError(server.Serve(listener))
+	}()
+
+	conn, err := grpc.NewClient(listener.Addr().String(), grpc.WithInsecure())
+	require.NoError(err)
+
+	client := NewClient(httppb.NewHTTPClient(conn))
+
+	w := &httptest.ResponseRecorder{}
+	r := &http.Request{
+		Header: map[string][]string{
+			"Upgrade": {"foo"}, // Make this look like a streaming request
+		},
+		Body: io.NopCloser(infiniteStream{}),
+	}
+
+	client.ServeHTTP(w, r)
+}
+
+type fooServer struct {
+	test.UnimplementedFooServer
+}
+
+func (fooServer) Foo(context.Context, *test.FooRequest) (*test.FooResponse, error) {
+	return &test.FooResponse{}, nil
+}
+
+type infiniteStream struct{}
+
+func (i infiniteStream) Read(p []byte) (n int, err error) {
+	return len(p), nil
+}
```

Comment:
I don't think we need this `fooServer` - or the `test` proto file at all
```suggestion
func TestRequestClientArbitrarilyLongBody(t *testing.T) {
	require := require.New(t)

	listener := bufconn.Listen(0)
	server := grpc.NewServer()
	httppb.RegisterHTTPServer(server, &httppb.UnimplementedHTTPServer{})

	go func() {
		require.NoError(server.Serve(listener))
	}()

	conn, err := grpc.NewClient(listener.Addr().String(), grpc.WithTransportCredentials(insecure.NewCredentials()))
	require.NoError(err)

	client := NewClient(httppb.NewHTTPClient(conn))

	w := &httptest.ResponseRecorder{}
	r := &http.Request{
		Header: map[string][]string{
			"Upgrade": {"foo"}, // Make this look like a streaming request
		},
		Body: io.NopCloser(infiniteStream{}),
	}

	client.ServeHTTP(w, r) // Shouldn't block forever reading the body
}

type infiniteStream struct{}

func (infiniteStream) Read(p []byte) (n int, err error) {
	return len(p), nil
}
```

Reply:
I forgot to unstage this... originally was trying to test this through a dummy service but ended up not using it